src/main/java/com/google/api/generator/gapic/composer/Composer.java

```diff
@@ -210,17 +209,48 @@ static List<GapicClass> prepareExecutableSamples(List<GapicClass> clazzes, Strin
               List<Sample> samples =
                   gapicClass.samples().stream()
                       .map(
-                          sample -> addRegionTagAndHeaderToSample(sample, apiShortName, apiVersion))
+                          sample ->
+                              addRegionTagAndHeaderToSample(
+                                  sample, parseDefaultHost(gapicClass.defaultHost()), apiVersion))
                       .collect(Collectors.toList());
               return gapicClass.withSamples(samples);
             })
         .collect(Collectors.toList());
   }
 
-  private static Sample addRegionTagAndHeaderToSample(
+  // Parse defaultHost for apiShortName. Need to account for regional default endpoints like
+  // "us-east1-pubsub.googleapis.com".
+  // Will keep backup name parsed from proto package for now.
+  @VisibleForTesting
+  protected static String parseDefaultHost(String defaultHost) {
+    String apiShortName;
+    String apiShortNameHelper;
+    // If the defaultHost is of the format "**.googleapis.com", take the name before the first
+    // period.
+    Pattern standardApiHost = Pattern.compile("\\.");
+    // If the defaultHost is of the format "**-**-**.googleapis.com", take the section before the
+    // first period and after the last dash to follow CSharp's implementation here:
+    // https://github.com/googleapis/gapic-generator-csharp/blob/main/Google.Api.Generator/Generation/ServiceDetails.cs#L70
```

Comment:
I think the solution in C# is much cleaner, we can do similar things i n Java as well. We don't have to declare each pattern as a separate variable, and we don't have to have a `apiShortNameHelper` as well. Something like
```
String apiShortName = split using Splitter with . and get the first one
apiShortName = split using Splitter with - and get the last one
```
Usually it's a good practice to check the length before we get it to prevent runtime exception, but we don't have to check it here since a well formatted defaultHost should not error out.

Reply:
Thank you - the Guava splitter makes it much easier to mimic C#. Updated!